rust-core/src/lib.rs

```diff
@@ -263,7 +266,7 @@ mod tests {
         assert_eq!(result.error_code, 1);
 
         // Clean up
-        lindos_result_free(result);
+        unsafe { lindos_result_free(result) };
```

Comment:
The `unsafe` wrapper around all the `free` function calls looks clumsy: It's repeated all over the place. Is there a better (more elegant, more DRY) way to do it?